R/lazy-select-query.R

```diff
@@ -94,6 +94,42 @@ update_lazy_select <- function(select, vars) {
   select
 }
 
+is_lazy_select_query_simple <- function(x,
+                                        ignore_group_by = FALSE,
+                                        select = c("projection", "trivial")) {
+  select <- arg_match(select, c("projection", "trivial"))
+  if (!inherits(x, "lazy_select_query")) {
+    return(FALSE)
+  }
+
+  if (!purrr::every(x$select$expr, is_symbol)) {
+    return(FALSE)
+  }
+
+  vars_prev <- op_vars(x$x)
+  if (select == "trivial" && !is_select_trivial(x$select, vars_prev)) {
```

Comment:
Put `is_select_trivial()` near this function?

Reply:
I added a comment and renamed `trivial` to `identity` which feels a bit clearer.